WordPress/Classes/ViewRelated/Comments/CommentModerationBar.swift

```diff
@@ -112,6 +118,26 @@ private extension CommentModerationBar {
         buttonStackViewTrailingConstraint.constant = horizontalPadding
     }
 
+    func setButtonForStatus() {
+        guard let comment = comment,
+              let commentStatusType = CommentStatusType.typeForStatus(comment.status) else {
+                  return
+              }
+
+        switch commentStatusType {
+        case .pending:
+            pendingTapped(pendingButton)
+        case .approved:
+            approvedTapped(approvedButton)
+        case .unapproved:
+            trashTapped(trashButton)
+        case .spam:
+            spamTapped(spamButton)
```

Comment:
one minor question that I should've raised in the previous PRs 😓 : I wonder if in the `pendingTapped` method and friends we could directly target the corresponding `UIButton` instead? Since I assume it's always guaranteed that `pendingTapped` is always associated with `pendingButton` and so on.

This is all so we could remove the need of parameters for these methods, i.e. `pendingTapped()`, `approvedTapped()`, and so on.

Reply:
Ah, yea. Nice catch. And done!